src/utils/bullet.utils.ts

```diff
@@ -0,0 +1,37 @@
+import { DAYS_IN_REPAYMENT_PERIOD, MONTHS_IN_YEAR } from "@consts";
+import { Ratio } from "@credix/credix-client";
+
+export const calculateMonthlyPayment = (principal: number, rate: Ratio, paymentPeriods: number) => {
```

Comment:
in the client we use 

```
export interface BulletLoanPeriodsGenerationConfig {
	principal: number;
	interest: Fraction;
	totalRepayments: number;
}

export const generateBulletLoanPeriods = (
	config: BulletLoanPeriodsGenerationConfig
): RepaymentPeriodConfig[] => {
	const interest = config.interest.apply(config.principal).toNumber();

	return Array(config.totalRepayments - 1)
		.fill(0)
		.map(() => ({
			principal: 0,
			interest,
		}))
		.concat({ principal: config.principal, interest });
};
```
for generating bullet loans. it's a lot more concise. (still rudimentary as it just keeps the decimals but that also happens here).
this interface and method are already exposed in the client so we can just use those